src/OrchardCore/OrchardCore.DisplayManagement/Shapes/DateTimeShapes.cs

```diff
@@ -113,6 +114,58 @@ public IHtmlContent TimeSpan(DateTime? Utc, DateTime? Origin)
                    : H["in a moment"];
     }
 
+    [Shape]
+    public IHtmlContent Duration(TimeSpan? timeSpan)
+    {
+        if (timeSpan == null)
+        {
+            return HtmlString.Empty;
+        }
+
+        var days = (int)timeSpan.Value.TotalDays;
+        var hours = timeSpan.Value.Hours;
+        var minutes = timeSpan.Value.Minutes;
+        var seconds = timeSpan.Value.Seconds;
+
+        var builder = new HtmlContentBuilder();
+
+        if (days > 0)
+        {
+            builder.AppendHtml(H.Plural(days, "{1} day", "{1} days", days));
+        }
+
+        if (hours > 0)
+        {
+            if (builder.Count > 0)
+            {
+                builder.AppendWhitespace();
+            }
+            builder.AppendHtml(H.Plural(hours, "{1} hour", "{1} hours", hours));
```

Comment:
You can't concatenate translated strings. It has to be a single one, which is this case doesn't seem possible because of pluralization (days could be one and hours multiple).

The only solution I could think of is to only keep the big part like "{0} days", or "{o} hours" and not concatenate.

Reply:
why wouldn't this work? each phrase can be translated separately and then build into html content. 

Here is an example

![image](https://github.com/user-attachments/assets/e2ecc2de-8f5d-434f-a8e0-8c785624c709)

Reply:
> why wouldn't this work

For one some people write from right to left ;) And pretty sure we can find languages where durations don't work this way too.